File: 1002-find-common-characters/1002-find-common-characters.cpp
class Solution {
public:
    vector<string> commonChars(vector<string>& words) {
        
        int min_len = 100;
        for (int i = 0; i < size(words); i++){
            min_len = min(min_len, int(words[i].length()));
        }

        vector<string> output;
        
        int i = 0;
        while(i < words[0].length()){
            char c = words[0][i];
            bool ok = true;

            for (int j = 0; j < size(words); j++){
                if(words[j].find(c) == string::npos){
                    ok = false;
                    break;
                }
            }
            if (ok){
                string s = "";
                s += c;
                output.push_back(string(1,c));
                for (int k = 0; k < size(words); k++){
                    auto pos = words[k].find(c);
                    words[k].erase(pos,1);
                }
            }
            else{
                i++;
            }
        }
        return output;
    }
};

File: 1002-find-common-characters/1002-find-common-characters.py
class Solution:
    def commonChars(self, words: List[str]) -> List[str]:
        output = []

        # iterate over character of the first word
        for c in words[0]:
            ok = True
            
            # check if c exists in all words
            for j in range(len(words)):
                if c not in words[j]:
                    ok = False
                    break
    
            if ok:
                output.append(c)

                # remove one occurence from each word
                for k in range(len(words)):
                    s = list(words[k])
                    s.remove(c)
                    words[k] = "".join(s)
                
        return output
            

               


        